sktime/forecasting/online_ensemble.py

```diff
@@ -0,0 +1,150 @@
+import numpy as np
+import pandas as pd
+from sktime.forecasting.base._base import DEFAULT_ALPHA
+from sktime.forecasting.compose._ensemble import EnsembleForecaster
+from sktime.forecasting.model_selection import SlidingWindowSplitter
+from sktime.utils.validation.forecasting import check_cv
+from sktime.utils.validation.forecasting import check_y
+
+
+class OnlineEnsembleForecaster(EnsembleForecaster):
+    """Online Updating Ensemble of forecasters
+
+    Parameters
+    ----------
+    ensemble_algorithm : ensemble algorithm
+    forecasters : list of (str, estimator) tuples
+    n_jobs : int or None, optional (default=None)
+        The number of jobs to run in parallel for fit. None means 1 unless
+        in a joblib.parallel_backend context.
+        -1 means using all processors.
+    """
+
+    _required_parameters = ["forecasters"]
+
+    def __init__(self, forecasters,
+                 ensemble_algorithm=None,
+                 n_jobs=None):
+
+        self.n_jobs = n_jobs
+        self.ensemble_algorithm = ensemble_algorithm
+        self.weights = np.ones(len(forecasters))/len(forecasters)
+
+#         if self.ensemble_algorithm.n != len(forecasters):
+#             raise ValueError("Number of Experts in Ensemble Algorithm \
+#                              doesn't equal number of Forecasters")
+
+        super(EnsembleForecaster, self).__init__(forecasters=forecasters,
+                                                 n_jobs=n_jobs)
+
+    def fit(self, y_train, fh=None, X_train=None):
+        """Fit to training data.
+
+        Parameters
+        ----------
+        y_train : pd.Series
+            Target time series to which to fit the forecaster.
+        fh : int, list or np.array, optional (default=None)
+            The forecasters horizon with the steps ahead to to predict.
+        X_train : pd.DataFrame, optional (default=None)
+            Exogenous variables are ignored
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+
+        self._set_y_X(y_train, X_train)
+        self._set_fh(fh)
+        names, forecasters = self._check_forecasters()
+        self._fit_forecasters(forecasters, y_train, fh=fh, X_train=X_train)
+        self._is_fitted = True
+        return self
+
+    def _fit_ensemble(self, y_new, X_new=None):
+        """Fits the ensemble by allowing forecasters to predict and
+           compares to the actual parameters.
+
+        Parameters
+        ----------
+        y_new : pd.Series
+            Target time series to which to fit the forecaster.
+        X_new : pd.DataFrame, optional (default=None)
+            Exogenous variables are ignored
+        """
+        if len(y_new) >= 1 and self.ensemble_algorithm is not None:
```

Comment:
I find that the code is more readable if that if/else statement is moved outside of the function into `self.update`, so:
```python
if condition:
   self._fit_ensemble(...)
```

Reply:
I agree, that makes sense, I'll move it there